dts/common/zephyr_thermistor.dtsi

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	/omit-if-no-ref/ thermistor_R25_10K_B_3974: thermistor-R25-10K-B-3974 {
```

Comment:
What is going on here? Why is this being added to `dts/common/`? This all looks like very hardware-specific property settings that don't belong here.

Reply:
Hi Marti - I'm not exactly sure where this should live.  This looked like a good place.  NTC thermistors can share similar characteristics so the table can be used in other places (why I chose `dts/common`), so another manufacturer can reference this same table.  Would creating a `dts/thermistor` folder be more appropriate?

Reply:
> Would creating a dts/thermistor folder be more appropriate?

I don't think so, no. Look at the other directories under `dts`. We currently have SoC specific DTSI files and bindings in here, besides the common directory which contains things that apply to virtually every .dts used by zephyr. Putting a random node in here that looks like a hardware-on-my-desk type thing (the `_R25_10K_B_3974` is a tell, for example, at least if I'm reading that correctly) doesn't make sense IMO.

> can be used in other places

where?

Reply:
IIRC, it's possible to have many device tree compatibles all reference the same binding, and if the table is const in DT, then it might be possible to even have 1 driver for serval different thermistor profiles too.

Marti us the expert there though, so I defer to his advice.

Reply:
> A search on digikey came up with 52 [thermistors](https://www.digikey.com/en/products/filter/temperature-sensors/ntc-thermistors/508?s=N4IgjCBcoBwGwGYqgMZQGYEMA2BnApgDQgD2UA2iAEwAMYVA7FSALrEAOALlCAMqcAnAJYA7AOYgAvsXhxkINJCx4ipCiAYIALDSRsQXHv2HipkyUA) that have the same electrical characteristics that can reference this same table.

Just to be clear, I fully recognize that you are the domain expert on thermistors here, and I believe you that this is a frequently seen piece of hardware.

> So being consistent with other *.dtsi files, I think this file should go under dts somewhere

However, I do not yet agree that this approach is consistent with other *.dtsi files, or that it should go under dts/. Let me try to explain and we'll see how we can get this done.

Take ns16550 UARTs for example. This is also a very common piece of hardware. As far as I know, we do not have a *.dtsi file with a "reference" ns16550 node inside it anywhere under dts/, and I would argue that we should not. Would you want that to be added to a bunch of devicetrees that don't need it, and then ripped out with `/omit-if-no-ref/` in 99.999...% of cases? I think that would be inappropriate. Reasoning by analogy, I think this current approach is also inappropriate.

> may be a new use case for a device tree node that's shared between thermistors, that's not part of an SOC/board.

The usual idiom is to provide example nodes in the `description:` field of the devicetree binding when there are common examples. That lets everyone who has that piece of hardware see a working example they can copy/paste into their SoC DTSI or board DTS. It really doesn't seem like this is any different.

>> can be used in other places
>
> where?

That is why I am asking you where specifically these "other" places are. If it's just board DTS files, for example, IMO we should go with the usual idiom of examples in the description. If this really is some sort of rare bird, however, I think it's fair to ask you to provide more examples of where it's going to show up that prove the point.

Reply:
Hi Marti - I'm discussing with my team on this.  You can disregard my most recent push moving this file out of `dts/common`.

Reply:
Hi @mbolivar-nordic, @cfriedt 

What do you think of defining the table in C code guarded by `DT_HAS_COMPAT_STATUS_OKAY` check?

Something like

zephyr_thermistor.c
```
#if DT_HAS_COMPAT_STATUS_OKAY(zephyr_thermistor_rt_table)
struct ntc_compensation THERMISTOR_NAME = [...];
#endif 
```

boards/arm/tdk_robokit1/tdk_robokit1-common.dtsi
```
	thermistor_R25_10K_B_3974: thermistor_R25_10K_B_3974 {
		compatible = "zephyr,thermistor-rt-table";
	};
	temp_sensor: ambient_temp_sensor {
		compatible = "zephyr,thermistor";
		io-channels = <&spi_adc 0>;
		thermistor = <&thermistor_R25_10K_B_3974>;
		r25-ohm = <10000>;
		pullup-mv = <3300>;
		pullup-ohm = <0>;
		pulldown-ohm = <10000>;
		connection-type = "NTC_CONNECTED_POSITIVE";
	};
```

Reply:
@asemjonovs I really can't say without knowing more about the places this is going to be used like I was asking about. The heart of my questions/concerns here are "why is this hardware so special that it's different than literally every other situation we have in the tree", which your most recent question doesn't answer for me. Make sense?

Reply:
@mbolivar-nordic - let me try to give a hypothetical example if that helps.

Company A wants to monitor their CPU temperature with an Epcos Thermistor (r25 10k ohm, B3974).
Company B wants to monitor the motor temperature with another NTC thermistor (r25 47k ohm, B4050).
A home enthusiast wants to monitor the temperature of their home using an Amertherm thermistor (r25 10k ohm, B3974)

Company B needs to define a resistance to temperature table for their thermistor. Company A and the home enthusiast's thermistor share the same electrical characteristics but use different boards.  They can copy each other's device tree node for the thermistor which is about 90 lines long for defining the table.  If I understand correctly, this is how the UART example you gave works.  Different boards may use the same UART and configure it with different or same clock frequencies and so on.  One line in the DTS configures this for the UART.

What I'm trying to accomplish with the `zephyr_thermistor.dtsi` is make a node defining this 90 line table available to share between boards.  If one of the company's want to swap to another NTC thermistor, they can update one line in their board's dts.

I'm Zephyrizing Linux's implementation of the [NTC Thermistor](https://github.com/torvalds/linux/blob/master/drivers/hwmon/ntc_thermistor.c#L72) but they define the table in C code instead of DTS.   

Does this help clarify?  We can sync up over discord if needed.

Reply:
> Does this help clarify and answer your question? We can sync up over discord if needed.

I think we probably better had. We should try to get this into dev-review this week if we can't figure out a solution by EOD wednesday as well, I think.